src/network/networkd-link.c

```diff
@@ -2619,16 +2619,33 @@ static int link_set_ipv6_mtu(Link *link) {
 }
 
 static int link_drop_foreign_config(Link *link) {
-        Address *address;
+        Address *address, *net_address;
         Route *route;
         Iterator i;
         int r;
+        bool skip_address;
 
         SET_FOREACH(address, link->addresses_foreign, i) {
                 /* we consider IPv6LL addresses to be managed by the kernel */
                 if (address->family == AF_INET6 && in_addr_is_link_local(AF_INET6, &address->in_addr) == 1)
                         continue;
 
+                if (link->network) {
+                        skip_address = false;
+                        LIST_FOREACH(addresses, net_address, link->network->static_addresses) {
```

Comment:
The bracket `{` after `LIST_FOREACH` seems not necessary.

Reply:
> Btw does this relate to the coding style item "Single-line `if` blocks"?

Yeah.

Reply:
so I should add them again?